### ISSUE TITLE
Improving usage of hashtags

### ISSUE THREAD

**Issue by thewhitetulip** [federicoiosue/Omni-Notes#101]
Currently hashtags aren't used effectively, they need to be coloured, making them visually distinct

**Comment by federicoiosue** [comment]
They actually are underlined to give advice to the user that he can interact with it with a double-touch action, as it is for other link types.

The current tags-categories duo is many to be used as in Google+ communities: they have a complementary goal and they works better when mixed.

**Comment by thewhitetulip** [comment]
Actually I have been using ON for a long time, and it was really few days after I installed the beta of 5.0 that I realized that hashtags can be double clicked, I was suggesting displaying hashtags in different colours at least. 

There can be many optimizations done to improve the usability of ON actually, first of all not everyone uses ON just to take notes, many use it as their TODO list, so they might skip the categories feature totally, so we may provide an option to use hashtags a bit properly.

**Comment by federicoiosue** [comment]
I'll keep the issue opened for future improvements based on your ideas

**Comment by thewhitetulip** [comment]
Improving hashtags means enabling the user to check which notes are tagged under a particular hashtag. 

Currently we have categories right? If I have a category called "Diary", it shows up in my navigation drawer and when I click on it I get to see the notes that are under that category.

But if I want to see notes tagged under '#MyThoughts' then I have to go to search -> type hashtag [or] search->tags->select tags

We can hit two birds with one stone by directly giving a tags option which we show in the search menu on the 'action bar, so the user doesn't need to go through search -> tags -> select tags and can directly click on the tags on the action bar which will cut through the hierarchy.

This will make the app versatile because we are giving the users two powerful options to group their notes into and two equally accessible ways to see the grouping.

If you don't want to show the tags by default then it can be put in the overflow menu, but I strongly feel that hashtags should be given equal status as the categories.

**Comment by sb56637** [comment]
> @thewhitetulip  I strongly feel that hashtags should be given equal status as the categories.

I agree. Hastags are extremely powerful, but it would be nice if I didn't have to filter the searches every time. Or maybe we could allow for saved searches?
